h3ndrk/hulk
# Add binary serialization to Communication

## Introduced Changes

What it says, this PR enables Communication to serialize binary messages from *any* given path. The `SerializeHierarchy` has been extended to support binary serialization. The communication server constructs binary messages and sends them to the client. To enable efficient image serialization, the `Image` type in this PR receives a special `Serialize` implementation s.t. it de-/serializes into/from JPEG RGB data. Test cases have been added to ensure correct execution.

## ToDo / Known Issues

None

## Ideas for Next Iterations (Not This PR)

None

## How to Test

- Execute the test cases (`types` and `communication` crates)
- Try to register a subscription with binary format and observe additional binary messages in [bincode](https://github.com/bincode-org/bincode) format according to `messages.rs`
